### PR TITLE
Replace `linked_port` with `linked_ports` in HugrView

### PR DESCRIPTION
**Pull request by aborgna-q**
Since we now support multiple links to the same port

### CODE REVIEW

File: src/hugr/view.rs [hunk 1]
```diff
@@ -67,7 +64,12 @@ pub trait HugrView {
     fn all_node_ports(&self, node: Node) -> NodePorts;
 
     /// Return node and port connected to provided port, if not connected return None.
-    fn linked_port(&self, node: Node, port: Port) -> Option<(Node, Port)>;
+    fn linked_ports(&self, node: Node, port: Port) -> PortLinks<'_>;
```
Comment (cqc-alec) [inline src/hugr/view.rs:67]: Fix docstring.

File: src/hugr/view.rs [hunk 2]
```diff
@@ -227,3 +226,43 @@ where
...
+
+/// Iterator over the links of a port
+#[derive(Clone)]
+pub struct PortLinks<'a> {
+    hugr: &'a Hugr,
+    links: crate::hugr::multiportgraph::PortLinks<'a>,
+}
+
+impl<'a> Iterator for PortLinks<'a> {
+    type Item = (Node, Port);
```
Comment (cqc-alec) [inline src/hugr/view.rs:238]: Wonder if this is worth a more general type definition (not in this PR though).

File: src/builder/build_traits.rs [hunk 3]
```diff
@@ -604,7 +604,7 @@ fn wire_up<T: Dataflow + ?Sized>(
         }
     }
 
-    if let Some((connected, connected_offset)) = base.hugr().linked_port(src, src_offset) {
+    if let Some((connected, connected_offset)) = base.hugr().linked_ports(src, src_offset).next() {
```
Comment (cqc-alec) [inline src/builder/build_traits.rs:607]: Should we be iterating over _all_ linked ports here?

**Review by cqc-alec — CHANGES_REQUESTED** [review]

File: src/builder/build_traits.rs (hunk 3 above)
Comment (aborgna-q) [reply src/builder/build_traits.rs:607]: This is for the explicit-copies builder. This part of the code is directly dropped in the next PR #140

File: src/hugr/view.rs (hunk 2 above)
Comment (aborgna-q) [reply src/hugr/view.rs:238]: The iterator story is quite cumbersome. Hopefully we can clean it up in the future.

**Review by cqc-alec — APPROVED** [review]

**Pull request merged**